joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -67,24 +67,23 @@ def init_sdf(self, robot):
                 continue
             if child.localName == 'joint':
                 jtype = child.getAttribute('type')
-                if jtype in ['gearbox', 'revolute2', 'ball', 'screw', 'universal', 'fixed']:
+                if jtype in ('gearbox', 'revolute', 'ball', 'screw', 'universal', 'fixed'):
                     continue
                 name = child.getAttribute('name')
                 self.joint_list.append(name)
+
+                # joint limits
                 if jtype == 'continuous':
                     minval = -math.pi
                     maxval = math.pi
                 else:
-                    # TODO actual limits
-                    minval = -math.pi  # xxx
-                    maxval = math.pi  # xxx
-                    # try:
-                    #     limit = child.getElementsByTagName('limit')[0]
-                    #     minval = float(limit.getAttribute('lower'))
-                    #     maxval = float(limit.getAttribute('upper'))
-                    # except:
-                    #     self.get_logger().warn('%s is not fixed, nor continuous, but limits are not specified!' % name)
-                    #     continue
+                    try:
+                        limit = child.getElementsByTagName('limit')[0]
+                        minval = float(limit.getElementsByTagName('lower')[0].firstChild.data)
+                        maxval = float(limit.getElementsByTagName('upper')[0].firstChild.data)
+                    except:
+                        self.get_logger().warn('%s is not fixed, nor continuous, but limits are not specified!' % name)
```

Comment:
My previous comment could help with this where we catch different types of exceptions and report different errors, but as it is currently, the exception might have been through by the string->float conversion and not because the `limit` tag is missing. If we want to keep using exceptions, I think we can catch `ValueError` for string->float conversion, but `getElementsByTagName` doesn't throw, so I think it might be best to check the return value.

Reply:
Since the rest of the codebase follows the same pattern, it's probably not worth the effort to check the result of `getElementsByTagName` here. I'll consider this resolved.